### PR TITLE
DOC: add mission statement

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

This is a (very) rough draft of a follow on to contents of history.rst which lays out what JDH understood Matplotlib to be when it started that lays out what we think Matplotlib should be going forward.

### CODE REVIEW

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:10]:
```suggestion
The Matplotlib developer community develops, maintains, and
supports Matplotlib and its extensions, providing plotting tools for 
the Scientific Python Ecosystem.  Our success is based on how effectively we
enable our users, across scientific, industrial, commercial, and educational
applications, to achieve their goals.  We believe that a diverse developer community creates
the best software, and we welcome anyone, regardless of experience, education,
gender, or nationality, who shares our mission to the community.
```

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:26]:
```suggestion
To achieve these goals simultaneously across the full gamut of the domains that depend on Matplotlib, 
we have adopted a model of a general core library that supports an ecosystem of
domain-specific extensions that inter-operate as much as practical.
```

**Review by jklymak — COMMENTED** [review]
Just a few wordsmithing suggestions.   Looks great!

File: doc/users/project/mission.rst
Comment (story645) [inline doc/users/project/mission.rst:19]:
```suggestion
* Be understandable and extendable
```
I'm not sure the scoping is helping here - like would an industry developer (like at Bloomberg) consider themselves an engineer or data scientist?

File: doc/users/project/mission.rst
Comment (story645) [inline doc/users/project/mission.rst:21]:
```suggestion
* Make common cross domain plots easy;
```
 standard is domain specific

File: doc/users/project/mission.rst
Comment (timhoffm) [inline doc/users/project/mission.rst:4]: Maybe split in two sentences: Matplotlib is… The Mission of the Matplotlib community is …

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:19]: I think yes, most of the people who we have worked with at Bloomberg were in an org with the word "Engineering" in the name.

Fundamentally Matplotlib is a tool for _doing_ science, data science, and engineering and I think it is important that we strive to be accessible to the practitioners of those things.

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst]: @timhoffm do you like Jody's edits?

File: doc/users/project/mission.rst
Comment (story645) [reply doc/users/project/mission.rst:19]: My concern is that folks who do science/data science/engineering/quantitative research methods wouldn't necessarily label themselves as scientists/data scientists/engineers- for example students, who I think are one of our largest user bases and a group we explicitly consider in tutorials/docs/outreach. 

What about separating the two and  tool/task centered framing ? (it still needs wordsmithing, but wanted to jot out something quick)
* Understandable to people^ creating data and scientific data visualizations  
* Extendable by developers who build data and scientific data visualization tools 

^students, researchers, developers, analysts, engineers - I'm not sure what the collective noun for this bucket is

File: doc/users/project/mission.rst
Comment (story645) [inline doc/users/project/mission.rst:10]: I worry explicitly listing some groups but not others - like race/ethnicity, religion/creed,  sexual orientation - might get us into "why those?" discussions. Maybe go more general here?

```suggestion
applications, to achieve their goals.  We believe that a diverse developer community creates
the best software, and we welcome anyone who shares our values, as described in the code of conduct, and our mission to the community.
```

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:19]: I think we can look at the numpy user demographics and use it as a rough guide for us (https://github.com/numpy/numpy-surveys/blob/master/images/2020NumPysurveyresults_community_infographic.pdf).   In the case of numpy, they had 90% of users have bachlor's degrees and only 22% are using the library for "studies" vs 69% who are using it for "work".    Based on this I do not think it is accurate to say students are one of our biggest user bases.

Further students are, by definition, still learning.  It is OK if things are not immediately understandable by students their task to to learn enough so that it is understandable!  I suspect that a majority of the students who are using Matplotlib are doing so because they are in training to be scientists, data scientists, and engineers so I am not worried about saying we are a tool for the role they are in training to do.

I do not see any tension between a) having the goal of being understood by people working with Matplotlib in a professional context and b) making sure our documentation/tutorials do a good job of address people studying to take those roles!

I do not want to split this in half because I think we should be both understandable and extendable by the people who are using Matplotlib to get stuff done.

I stand by the correct collective noun being "scientists, data scientists, and engineers".

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:21]: fair, but I think that "standard" is still clear and "common cross domain" is too jargonny.

Comment (jklymak) [reply doc/users/project/mission.rst:20]: I think "cross domain" is pretty jargony, and I'm not even sure we didn't coin it ourselves, as Matplotlib is the only context I have heard it used for.

**Review by story645 — DISMISSED** [review]
placeholder for longer response or discussion on call

File: doc/users/project/mission.rst
Comment (dopplershift) [reply doc/users/project/mission.rst:20]: Why not just "common plots"?

File: doc/users/project/mission.rst
Comment (dopplershift) [reply doc/users/project/mission.rst:18]: My prior here is: I think students studying in the sciences are by definition scientists. They are people who are seeking to better understand the world through the application of the scientific method.

I'm willing to be convinced by data or an abundance of opinions to the contrary. I don't think myself 20 years ago would have been turned off by that language, but then again I'm not a part of any group that we struggle to represent here.

Calling out both "scientist" and "data scientist" feels redundant to me, though.

Comment (story645) [reply doc/users/project/mission.rst:18]: > My prior here is: I think students studying in the sciences are by definition scientists
This might be just diss depression, but like I don't really feel like I could call myself a scientist (I cringe at an interview where I did) or an engineer because those aren't official titles I've held. I also am not sure what we gain by including a physicist doing art restoration and excluding an art historian making the exact same heatmaps (and sometimes both on the same interdisciplinary team ). (And yes you may consider the art historian a scientist or data scientist in this context, but I dunno that they would) This is why I was suggesting language that expressed what folks do (create scientific visualizations, build scientific visualization tools) rather than the ways in which they identify.  

> Calling out both "scientist" and "data scientist" feels redundant to me, though.
Yeah, and I don't love it, but the carry w/ 'em distinct notions of the type of analysis folks are doing. And I've been thinking lately about where teachers/teaching track professors would slot in (folks using matplotlib primarily to teach science and math, which is doing science and also is a loaded debate) & I swear I'm not trying to split hairs or be pedantic, but I'm trying to unpack what it is we're really trying to scope for here. 

>I do not see any tension between a) having the goal of being understood by people working with Matplotlib in a professional context and b) making sure our documentation/tutorials do a good job of address people studying to take those roles!

It's not that they're in conflict, it's that the mission statement to me reads like a prioritization of a) in a way that I worry could feel intimidating/like matplotlib is not for b).  Which is why I suggested a split, b/c we want the library to be understandable by a more general audience than we expect to build new tools with it - is why we have a mix of different types of docs.

Comment (jklymak) [reply doc/users/project/mission.rst:18]: If we are defining who Matplotlib will be comprehensible to, perhaps this could be rephrased based on what level of knowledge we will be try to be comprehensible to, not necessarily domains or job titles. I agree that most scientists and engineers will have that level of knowledge, but no need to exclude a knowledgeable art historian. (Conversely, I do think the mission statement should specifically say we are a scientific visualization library; you are welcome to use it for art, image manipulation, or architecture, but don't ask us to add features tuned for those disciplines as it is out of scope.)

Defining these minimum qualifications is a little harder than tautologically saying that a scientific visualization library should be understandable to scientists, but perhaps more practically useful when making decisions about how in the weeds the docs or API need to go.  Maybe something like "understandable to people with basic python programming skills and familiarity with numerical processing techniques".

Comment (efiring) [reply doc/users/project/mission.rst:18]: I think Jody's suggestion or a variation of it is a good alternative.  Suggestion: "understandable to people familiar with data processing techniques and with the Python programming language."  Or "understandable and extendable by..."

File: doc/users/project/mission.rst
Comment (efiring) [reply doc/users/project/mission.rst:20]: Yes, "common plots".

**Review by efiring — CHANGES_REQUESTED** [review]
It's close to ready, but I have some suggestions for consideration.

**Comment by tacaswell** [comment]
I think this is ready for a final review.  I would like to get put this on the agenda for next weeks dev call.

I am going to assert the merge conditions for this is consensus from the steering council.

**Comment by tacaswell** [comment]
The lens I think we should use to read this through is answering the questions:

 - What is the standard we use to evaluate if a proposal for a feature / removal / activity is "in-scope" for the project, if the proposals advances the project, and if so by how much so (we can weight the cost/benefit)?
 - What is the thing we are doing as a group that we would not be able to do as individuals?

I think this statement for our own internal guidance for "what it is we do here" and for people who what to start working on the project, not an external facing document for users to determine if using this library is "for them" or not.  Nor does this need to enumerate every activity we are going go do in support of the mission (keeping the CI up is critical, but also has no place in the mission statement!).  Similarly, a clean and consistent API or complete well written documentation are things we do to accomplish the mission, but are not the purpose in and of themselves.

Being a historically and currently volunteer organization we have all brought our own intrinsic motivations for why we do things and each make decisions about what we spend our time on to align with those internal motivations, however now that we have funding (and plan to get more) we need to be able to articulate what our mission is.

I was motivated to articulate this now (well, last Nov) by training from CZI about how to develop and maintain an organizational-level strategic plan.  The first prerequisite for any of that work is to agree what the mission is!

File: doc/users/project/mission.rst
Comment (timhoffm) [inline doc/users/project/mission.rst:12]:
```suggestion
conduct.
```
Can be left out. This avoids the spitting of the sentence by "as described ..." and is implied through the context of the previous sentence.

File: doc/users/project/mission.rst
Comment (timhoffm) [inline doc/users/project/mission.rst:20]: Do we want to say something about web and/or interactivity? These are not our strong sides right now, but we should discuss whether that should be our goal.

File: doc/users/project/mission.rst
Comment (efiring) [inline doc/users/project/mission.rst:7]: domain -> domains

File: doc/users/project/mission.rst
Comment (efiring) [inline doc/users/project/mission.rst:27]: as -> a

File: doc/users/project/mission.rst
Comment (efiring) [reply doc/users/project/mission.rst:20]: GUI application implies interactivity, though the word could be inserted to make it explicit.  I think that better interactivity in web apps like Jupyter notebook is important.  Maybe modify to "...application development, standalone and on the web."

**Review by efiring — CHANGES_REQUESTED** [review]
After fixing a couple typos and addressing Tim's comments, it will be ready.

**Review by efiring — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:6]: This reads like we are saying we are successful, rather than what I think you mean, which is that we consider it a success if we can do these things.   Maybe 

```suggestion
Ecosystem.  We judge our success based on how effectively we enable our users, across
```

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:19]:
```suggestion
* Produce high-quality raster and vector format outputs;
```

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:21]: "standalone" seems logically orphaned between two embeddings?  Maybe:
```suggestion
* Be usable as a stand-alone graphical user interface, and to be embeddable in applications, both native and on the web.
```

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:6]: That is indeed what I meant.  This wording is better.

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:21]: I agree this is a better wording.

**Comment by jklymak** [comment]
`history` isn't defined?  Otherwise, this looks good to me!

**Review by dopplershift — APPROVED** [review]

File: doc/users/project/mission.rst
Comment (tacaswell) [inline doc/users/project/mission.rst:7]:
```suggestion
academic, industrial, commercial, and educational domains, to achieve their
```

@noatamir pointed out that "scientific" is the odd-one-out in this sentence and "academic" is a better word for what is meant.

Comment (efiring) [reply doc/users/project/mission.rst:7]: The suggested change leaves out NASA, NOAA, USGS, etc.  Is there a concise way to include them?  Or are we trying to do too much enumeration of specific categories?  Maybe just leave out the whole list, and leave it at "enable our users to achieve their goals".  That leaves the domain specification to "Scientific Python Ecosystem"; maybe that is enough.

Comment (dopplershift) [reply doc/users/project/mission.rst:7]: I'm also not sure in that context what the difference between "academic" and "educational" is.

Comment (tacaswell) [reply doc/users/project/mission.rst:7]: As someone who sits in a institution that is in that categories (in the US they tend to be called Federally Funded Research and Development Centers (FFRDC)), we are used to the ambiguity and will self-identify as whichever of the other categories suits us best at the time ;)  We could reference FFRDC but that would miss our peer institutions in other countries (and is _super_ jargonny).

There is some aspect of on-label and off-label uses of Matplotlib.  Our heritage is for scientific plotting (going back to John's EKG plots) and a majority of our developers are trained in some sort of science, and a bulk of the (maybe over) specialized plotting we have is driven by the needs of scientific visualization.  That we have also been used by a (wide) variety of people outside of the domains that our developers work in is a testament to the library and is great!  One of the best things about open source is that our users do things with the code that the core developers could not have imagined.  However, we can not support all domains equally; part of it is just the expertise we have in the core team (I think mpl-finance is a good example here that is _much_ better off with Daniel as a stand-alone project) and because there may be conflicting needs/expectations and some use is should be considered "intended" and some "we are happy it works".

Some of this is very easy to do: "please implement an email client" -> "no, obviously out of scope".  

Some is a bit harder, but still easy: "This API makes it hard to make (intentional) art with Matplotlib " -> "cool you are making art, but the purpose is to be a plotting library so we will not change the API"

Some are super subtle : "if I drop integer coded catagoricals into imshow and use interpolation I get fractional values" -> well, that is a bit of a miss-use of `imshow` because it implicitly assumes the values are continuous not discrete.  do we need a new function, make imshow (yet more) complex,  or is no-norm the solution here?

That is all a very long winded way of saying I think that it is important that we scope the intended application down from "anyone who installs the library and has a story about how they want to use it".

Another aspect that I want to make sure we do not lose is a sense that Matplotlib is always (well intended to be) used as a means to and end, not and end it self.   There is some reason (be it understanding some science measurement, scheduling operations, looking at revenue over time, looking at survey results, looking at market data, ....) that the user is making a plot.  Our goal is not to be the perfect platonic plotting library, but rather to be a tool in service of whatever it is our users are actually doing (I also think that this is the case for a vast majority of software).

Put another, to paraphrase Charity Majors "~9s~ design does not matter if users are not happy" .  We can not make all users happy all of the time so we should identify which users / applications we intend to make happy to help us with triage / decision making.

Comment (tacaswell) [inline doc/users/project/mission.rst:7]:
```suggestion
Ecosystem.  We judge our success based on how effectively we enable our users to achieve their
```

Comment (efiring) [reply doc/users/project/mission.rst:7]: "enable our users to solve problems in their scientific, educational, and operational domains."

File: doc/users/project/mission.rst
Comment (jklymak) [inline doc/users/project/mission.rst:5]:
```suggestion
and its extensions to provide data visualization tools for the Scientific Python
```

File: doc/users/project/mission.rst
Comment (oscargus) [inline doc/users/project/mission.rst:18]:
```suggestion
* Support all users of the Scientific Python ecosystem;
```
Link to https://scientific-python.org/ ? (Which I cannot really remember how to do.)

File: doc/users/project/mission.rst
Comment (tacaswell) [inline doc/users/project/mission.rst:19]: insert something about publication quality + latex?

Comment (anntzer) [inline doc/users/project/mission.rst:19]: Sorry for dropping in late in the discussion (thanks to @story645 for pointing me to it as well). Perhaps it is worth mentioning "publication quality plots" in there as well?

**Comment by tacaswell** [comment]
- enable our users to solve problems in their scientific, educational, and operational domains. 
 - enable our users to solve problems in their research, educational, and operational domains. 
 - enable our users to visualize their data in ....
- practical real world data analysis & research 

xref to pandas mission

> "pandas aims to be the fundamental high-level building block for doing practical, real world data analysis in Python. Additionally, it has the broader goal of becoming the most powerful and flexible open source data analysis / manipulation tool available in any language."

File: doc/users/project/mission.rst
Comment (tacaswell) [reply doc/users/project/mission.rst:17]: Produce publication-quality output in both raster and vector format outputs;

Comment (tacaswell) [reply doc/users/project/mission.rst:17]: Produce publication-quality output in both raster and vector formats

Comment (tacaswell) [reply doc/users/project/mission.rst:17]: Produce high-quality raster and vector format outputs suitable for publication

**Review by greglucas — APPROVED** [review]

**Review by dopplershift — APPROVED** [review]
I'm sure there's plenty of improvement to make this more of a "mission" statement, but I'm happy with how it reads now.

**Review by timhoffm — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Review by anntzer — APPROVED** [review]

**Review by story645 — APPROVED** [review]

**Comment by QuLogic** [comment]
Does no-one want to hit the button; might be the most approvals ever? I don't see anything in the notes, so I'm going to do it.

**Pull request merged**
